src/polyswarm_api/api.py

```diff
@@ -669,9 +669,38 @@ def download_id(self, out_dir, instance_id):
 
         return artifact
 
-    def sandbox(self, instance_id):
-        logger.info('Sandboxing %s', instance_id)
-        return resources.SandboxTask.create(self, artifact_id=instance_id).result()
+    def sandbox(self, instance_id, sandbox):
+        logger.info('Sandboxing %s in %s', instance_id, sandbox)
+        return resources.SandboxTask.create(self, artifact_id=instance_id, sandbox=sandbox).result()
+
+    def sandbox_file(self, artifact, sandbox, artifact_type=resources.ArtifactType.FILE, artifact_name=None):
+        logger.info('Sandboxing file in %s', sandbox)
+        artifact_type = resources.ArtifactType.parse(artifact_type)
+        # TODO This is a python 2.7 check if artifact is a file-like instance, consider changing
+        #  to isinstance(artifact, io.IOBase) when deprecating 2.7 and implementing making LocalHandle
+        #  inherit io.IOBase, although this will change the method delegation logic in the resource
+        if hasattr(artifact, 'read') and hasattr(artifact.read, '__call__'):
+            artifact = resources.LocalArtifact.from_handle(self, artifact, artifact_name=artifact_name or '',
+                                                           artifact_type=artifact_type)
+        elif isinstance(artifact, string_types):
+            if artifact_type == resources.ArtifactType.FILE:
+                artifact = resources.LocalArtifact.from_path(self, artifact, artifact_type=artifact_type,
+                                                             artifact_name=artifact_name)
+            elif artifact_type == resources.ArtifactType.URL:
+                artifact = resources.LocalArtifact.from_content(self, artifact,
+                                                                artifact_name=artifact_name or artifact,
+                                                                artifact_type=artifact_type)
```

Comment:
Is there a 3rd option between ArtifactType.FILE or ArtifactType.URL ?

Reply:
Not right now afaik, there might be in the future though, but it will involve bigger changes. This is the same code that we use to submit files for scanning, it is just copy/pasted for that matter.